src/coreclr/jit/importercalls.cpp

```diff
@@ -3176,6 +3179,18 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                             retNode = gtNewIconNode(
                                 (info.compCompHnd->getClassAttribs(hClass) & CORINFO_FLG_BYREF_LIKE) ? 1 : 0);
                             break;
+                        case NI_System_Type_get_IsPrimitive:
+                        {
+                            retNode        = gtNewFalse();
+                            uint32_t attrs = info.compCompHnd->getClassAttribs(hClass);
+                            if (((attrs & CORINFO_FLG_GENERIC_TYPE_VARIABLE) == 0) &&
+                                (info.compCompHnd->getTypeForPrimitiveNumericClass(hClass) != CORINFO_TYPE_UNDEF))
+                            {
+                                retNode = gtNewTrue();
```

Comment:
Return false for the negative case?

Reply:
It did return false for negative case, but I guess the code shape I used wasn't obvious, so changed it.